app/src/main/java/com/twilio/video/app/sdk/RoomFactory.kt

```diff
@@ -0,0 +1,143 @@
+package com.twilio.video.app.sdk
+
+import android.content.Context
+import android.content.SharedPreferences
+import com.twilio.androidenv.Env
+import com.twilio.video.AudioCodec
+import com.twilio.video.BandwidthProfileMode
+import com.twilio.video.BandwidthProfileOptions
+import com.twilio.video.ConnectOptions
+import com.twilio.video.EncodingParameters
+import com.twilio.video.G722Codec
+import com.twilio.video.H264Codec
+import com.twilio.video.IsacCodec
+import com.twilio.video.NetworkQualityConfiguration
+import com.twilio.video.NetworkQualityVerbosity
+import com.twilio.video.OpusCodec
+import com.twilio.video.PcmaCodec
+import com.twilio.video.PcmuCodec
+import com.twilio.video.Room
+import com.twilio.video.Video
+import com.twilio.video.VideoBandwidthProfileOptions
+import com.twilio.video.VideoCodec
+import com.twilio.video.Vp8Codec
+import com.twilio.video.Vp9Codec
+import com.twilio.video.app.data.Preferences
+import com.twilio.video.app.data.api.TokenService
+import com.twilio.video.app.util.EnvUtil
+
+class RoomFactory(
+    private val context: Context,
+    private val sharedPreferences: SharedPreferences,
+    private val tokenService: TokenService
+) {
+
+    suspend fun newInstance(
+        identity: String,
+        roomName: String,
+        isNetworkQualityEnabled: Boolean,
+        roomListener: Room.Listener
+    ): Room {
+
+            setSdkEnvironment(sharedPreferences)
+            val token = tokenService.getToken(identity, roomName)
+            val enableInsights = sharedPreferences.getBoolean(
+                    Preferences.ENABLE_INSIGHTS,
+                    Preferences.ENABLE_INSIGHTS_DEFAULT)
+
+            val enableAutomaticTrackSubscription: Boolean = getPreferenceByKeyWithDefault(
+                    Preferences.ENABLE_AUTOMATIC_TRACK_SUBSCRIPTION,
+                    Preferences.ENABLE_AUTOMATIC_TRACK_SUBSCRIPTION_DEFAULT)
+
+            val enableDominantSpeaker: Boolean = getPreferenceByKeyWithDefault(
+                    Preferences.ENABLE_DOMINANT_SPEAKER,
+                    Preferences.ENABLE_DOMINANT_SPEAKER_DEFAULT)
```

Comment:
@Alton09 Can you confirm that the these default values are consistent with the mobile browser recommended settings with BWP collaboration

https://www.twilio.com/docs/video/tutorials/developing-high-quality-video-applications